src/main/java/bio/terra/workspace/service/datareference/flight/CreateDataReferenceStep.java

```diff
@@ -0,0 +1,57 @@
+package bio.terra.workspace.service.datareference.flight;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.utils.FlightUtils;
+import bio.terra.workspace.db.DataReferenceDao;
+import java.util.UUID;
+import org.openapitools.jackson.nullable.JsonNullable;
+import org.springframework.http.HttpStatus;
+
+public class CreateDataReferenceStep implements Step {
+
+  private DataReferenceDao dataReferenceDao;
+
+  public CreateDataReferenceStep(DataReferenceDao dataReferenceDao) {
+    this.dataReferenceDao = dataReferenceDao;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext) throws RetryException {
+    FlightMap inputMap = flightContext.getInputParameters();
```

Comment:
This FlightMap should have the actual request (a CreateDataReferenceRequestBody) stored under the key JobMapKeys.REQUEST, which gets populated by the JobBuilder. Instead of creating a new custom key for each field of the request, it might be easier to retrieve the request and use its fields directly. You'd still need a few custom fields for info that wasn't in the request though, like workspaceId or referenceId.